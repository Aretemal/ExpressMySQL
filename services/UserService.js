import Post from "../models/Post.js";
import User from "../models/User.js";

class UserService{
    async create(user){
        const createdPost = await User.create(user);
        return createdPost;
    }
    async getOne(id) {
        if (!id) {
            throw new Error('Id не указан')
        }
        const user = await User.findByPk(id);
        return user;
    }
    async getStatus(id) {
        if (!id) {
            throw new Error('Id не указан')
        }
        const user = await User.findByPk(id);
        if(!user){
            throw new Error('Пользователь не найден')
        }
        const status = user.status;
        return status;
    }
}
export default new UserService()
