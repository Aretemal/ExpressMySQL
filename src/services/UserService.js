import User from '../models/user.js';
import { ResponseObjectJSON } from '../utils/creatorJSON.js';

class UserService {
  async getOne(id) {
    if (!id) {
      throw new Error('Id не указан');
    }
    const user = await User.findByPk(id);
    return ResponseObjectJSON.render({
      user,
      resultCode: 0,
    });
  }

  async getStatus(id) {
    if (!id) {
      throw new Error('Id не указан');
    }
    const user = await User.findByPk(id);
    if (!user) {
      throw new Error('Пользователь не найден');
    }
    const { status } = user;
    return ResponseObjectJSON.render({
      status,
      resultCode: 0,
    });
  }
}
export default new UserService();
