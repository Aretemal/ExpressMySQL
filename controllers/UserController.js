import UserService from '../services/UserService.js';

class UserController {
  async create(req, res) {
    try {
      const post = await UserService.create(req.body);
      res.json(post);
    } catch (e) {
      res.status(500).json(e);
    }
  }

  async getStatus(req, res) {
    try {
      const status = UserService.getStatus(req.params.id);
      res.json(status);
    } catch (e) {
      res.status(500).json(e);
    }
  }

  async getOne(req, res) {
    try {
      const user = await UserService.getOne(req.params.id);
      return res.json(user);
    } catch (e) {
      res.status(500).json(e);
    }
  }
}
export default new UserController();
