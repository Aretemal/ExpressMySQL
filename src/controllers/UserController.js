import UserService from '../services/UserService.js';
import fullUrlCreator from '../utils/fullUrlCreator.js';
import UserSerializers from '../serializers/UserSerializers.js';

class UserController {
  async getOne(req, res, next) {
    const user = await UserService.getOne(req.params.id);
    const serializer = new UserSerializers({
      attributes: user, id: req.user.id, type: 'User', link: fullUrlCreator(req),
    });
    req.serializer = serializer;
    next();
  }

  async getAllUsers(req, res, next) {
    const data = await UserService.getAllUsers(req.user.id, req.query);
    const serializer = new UserSerializers({
      attributes: data, id: req.user.id, type: 'Users', link: fullUrlCreator(req),
    });
    req.serializer = serializer;
    next();
  }
}
export default new UserController();
