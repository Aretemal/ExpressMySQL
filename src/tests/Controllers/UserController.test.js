/* eslint-disable no-undef */
import { jest } from '@jest/globals';
import UserController from '../../controllers/UserController';
import UserService from '../../services/UserService';

jest.fn('UserService');
const req = {
  params: {
    id: 1,
  },
};
const res = {
  dataJS: null,
  json: (data) => {
    res.dataJS = data;
  },
};

describe('User Controller :', () => {
  describe('getOne :', () => {
    test('should return info one user', async () => {
      function next() {}
      UserService.getOne = (id) => ({
        userId: id, login: 'Artem', firstName: 'Artem', lastName: 'Novik',
      });

      await UserController.getOne(req, res, next);

      expect(req.serializer)
        .toMatchSnapshot();
    });
  });

  describe('getAllUsers :', () => {
    test('should return array of users and count of users', async () => {
      function next() {}
      UserService.getOne = (id) => ({
        users: [{
          userId: id,
          login: 'Artem',
          firstName: 'Artem',
          lastName: 'Novik',
        }],
        countOfUsers: 5,
      });

      await UserController.getOne(req, res, next);

      expect(req.serializer)
        .toMatchSnapshot();
    });
  });
});
