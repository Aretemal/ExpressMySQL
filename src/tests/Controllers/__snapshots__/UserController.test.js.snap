// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Controller : getAllUsers : should return array of users and count of users 1`] = `
UserSerializer {
  "resource": {
    "countOfUsers": 5,
    "users": [
      {
        "firstName": "Artem",
        "lastName": "Novik",
        "login": "Artem",
        "userId": 1,
      },
    ],
  },
}
`;

exports[`User Controller : getOne : should return info one user 1`] = `
UserSerializer {
  "resource": {
    "firstName": "Artem",
    "lastName": "Novik",
    "login": "Artem",
    "userId": 1,
  },
}
`;
